### DISCUSSION TITLE
Change MIME type coming from S3

### DISCUSSION THREAD

**Discussion by dav-pascual** [nginxinc/nginx-s3-gateway#69] [Q&A] [answered]
I would like to use this proxy to list files in my S3. 
However some of the files are in format such as YAML, or simply gzipped plain text which the browser downloads instead of displaying them.

Would it be possible to set the MIME type according to mime.types and default_type instead of the type coming from S3 request?
Thanks

**Comment by dekobon** [comment, marked as answer]
I believe that the gateway is just passing along the MIME type as provided by S3. Typically, the preferred approach is to set the correct mime type by setting the  `Content-Type` header in S3 and then the gateway will pass it along in the same way. The same is true for the `Content-Disposition` header which informs the browser to either display or download a file.

See the AWS docs [here](https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObject.html).

Also, you could modify the headers by using one of the NGINX `proxy_*` directives to modify the HTTP headers.

**Reply by dav-pascual** [reply]
Many thanks @dekobon for your answer :)
I was doing a migration from on prem and wondered if I could keep setting the MIME types at Nginx level, but I suppose that doing it at S3 level will be better!

**Reply by ehginanjar** [reply]
@dav-pascual would you be so kind to tell me how to do that at S3 level? I'm using huawei obs at the moment.

**Reply by ehginanjar** [reply]
nvm, already found it.
